src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -436,6 +437,9 @@ def select(self, cols: List[Expression]) -> "SelectStatement":
             len(cols) == 1
             and isinstance(cols[0], UnresolvedAlias)
             and isinstance(cols[0].child, Star)
+            and not cols[0].child.expressions
+            # df.select("*") doesn't have the child.expressions
+            # df.select(df["*"]) has the child.expressions
```

Comment:
Somehow I feel these two comment lines should go above `and not cols[0].child.expressions`

Reply:
I'm good with either way and slightly prefer below it because the logic expressions are not interrupted by this comment.